server/game/GameObject.js

```diff
@@ -23,8 +23,18 @@ class GameObject {
     }
 
     getEffects(type) {
-        let filteredEffects = this.effects.filter(effect => effect.type === type);
-        return filteredEffects.map(effect => effect.getValue(this));
+        let suppressedEffects = this.getSuppressedEffects();
+        if(type) {
+            let filteredEffects = this.effects.filter(effect => effect.type === type && !suppressedEffects.includes(effect));
+            return filteredEffects.map(effect => effect.getValue(this));
+        }
+        return this.effects.filter(effect => !suppressedEffects.includes(effect));
```

Comment:
I don't really like that you're returning an array of raw effects when there's no type specified....
I don't know whether we ever should call getEffects with no type, but I think should be at least mapping them to their values

Reply:
In that case I would prefer that we call this function getEffectValues() and have a separate function for getEffects().  With suppressed effects I need a function that returns all of the effects minus any suppressed effects.  Unfortunately this function was getting in the way of that as it was named getEffects().

Reply:
Actually I'll add a new function `getRawEffects()` to return the unsupressed raw effects.

Reply:
That's fine - I think generally it would be better to avoid getRawEffects() as much as possible.  It will make future refactors of the effects system much easier...